From 2ca719a499735d865b3dacd049d562bad16c4aff Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 24 Jul 2024 10:57:42 +0200
Subject: [PATCH 1041/1695] clk: stm32mp25: fix RIF configuration checking

Semaphore mode is ignored if the CID filtering is disabled. Do check
that the CID filtering is enabled before checking that the semaphore
configuration is correct in stm32_rcc_has_access_by_id().

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git d69f8354a5acfafd9b2fa07a986f28719ec4916d clk: stm32mp25: fix RIF configuration checking]
Change-Id: Id1f68ce9be6c1423bdcd44819f63d96f32530441
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/395074
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/drivers/clk/clk-stm32mp25.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/drivers/clk/clk-stm32mp25.c b/core/drivers/clk/clk-stm32mp25.c
index 0eb41ec46..44f13ca65 100644
--- a/core/drivers/clk/clk-stm32mp25.c
+++ b/core/drivers/clk/clk-stm32mp25.c
@@ -844,6 +844,9 @@ static bool stm32_rcc_has_access_by_id(int id)
 
 	cid_reg_value = io_read32(rcc_base + RCC_R0CIDCFGR + 0x8 * id);
 
+	if (!(cid_reg_value & RCC_CIDCFGR_CFEN))
+		return true;
+
 	/*
 	 * First check conditions for semaphore mode, which does not
 	 * take into account static CID.
@@ -853,9 +856,6 @@ static bool stm32_rcc_has_access_by_id(int id)
 		return cid_reg_value & BIT(master + RCC_SEMWL_SHIFT);
 	}
 
-	if (!(cid_reg_value & RCC_CIDCFGR_CFEN))
-		return true;
-
 	/*
 	 * Coherency check with the CID configuration
 	 */
-- 
2.47.2

